samplers/aws/xray/client.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xray
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+type xrayClient struct {
+	// http client for sending unsigned proxied requests to the collector
+	httpClient *http.Client
+
+	endpoint *url.URL
+}
+
+// newClient returns an HTTP client with proxy endpoint
+func newClient(d string) *xrayClient {
+	endpoint := "http://" + d
+
+	endpointURL, err := url.Parse(endpoint)
+	if err != nil {
+		globalLogger.Error(err, "unable to parse endpoint from string")
+	}
+
+	return &xrayClient{
+		httpClient: &http.Client{},
+		endpoint:   endpointURL,
+	}
+}
+
+// getSamplingRules calls the collector(aws proxy enabled) for sampling rules
+func (p *xrayClient) getSamplingRules(ctx context.Context) (*getSamplingRulesOutput, error) {
+	statisticsByte, err := json.Marshal(getSamplingRulesInput{})
+	if err != nil {
+		return nil, err
+	}
+	body := bytes.NewReader(statisticsByte)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, p.endpoint.String()+"/GetSamplingRules", body)
```

Comment:
Store `p.endpoint.String()+/GetSamplingRules` (and targets later) in the struct rather than the endpoint to avoid recomputing this every time

Reply:
This is a struct that won't be initialized many times so having 2 fields is basically no overhead, while computing a string on every poll is overhead.